### PR TITLE
FEATURE: More flexible admin plugin config nav definition

### PR DESCRIPTION
**Pull request by martin-brennan**
This commit changes the API for registering the plugin config
page nav configuration from a server-side to a JS one;
there is no need for it to be server-side.

It also makes some changes to allow for 2 different ways of displaying
navigation for plugin pages, depending on complexity:

* TOP - This is the best mode for simple plugins without a lot of different
  custom configuration pages, and it reuses the grey horizontal nav bar
  already used for admins.
* SIDEBAR - This is better for more complex plugins; likely this won't
  be used in the near future, but it's readily available if needed

There is a new AdminPluginConfigNavManager service too to manage which
plugin the admin is actively viewing, otherwise we would have trouble
hiding the main plugin nav for admins when viewing a single plugin.

### CODE REVIEW

File: app/assets/javascripts/admin/addon/components/admin-plugin-config-page.gjs [hunk 1]
```diff
@@ -1,19 +1,19 @@
 import Component from "@glimmer/component";
 import { inject as service } from "@ember/service";
+import HorizontalOverflowNav from "discourse/components/horizontal-overflow-nav";
+import NavItem from "discourse/components/nav-item";
 import i18n from "discourse-common/helpers/i18n";
+import I18n from "discourse-i18n";
```
Comment (Drenmi) [inline app/assets/javascripts/admin/addon/components/admin-plugin-config-page.gjs:6]: 🤯 

How did we end up here? Should we have a TODO to consolidate this?

File: app/assets/javascripts/admin/addon/components/admin-plugin-config-page.gjs [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
 import AdminPluginConfigArea from "./admin-plugin-config-area";
 
 export default class extends Component {
   @service currentUser;
-
-  get configNavRoutes() {
-    return this.args.plugin.configNavRoutes || [];
-  }
+  @service adminPluginConfigNavManager;
```
Comment (Drenmi) [inline app/assets/javascripts/admin/addon/components/admin-plugin-config-page.gjs:11]: Sounds a bit like word salad. 😅 Not sure if we can cut out any of the words here?

**Review by Drenmi — APPROVED** [review]
Conceptually makes sense to me. Did my best to review the code. 🙂

**Comment by Drenmi** [comment]
> TOP - This is the best mode for simple plugins without a lot of different
custom configuration pages, and it reuses the grey horizontal nav bar
already used for admins.
SIDEBAR - This is better for more complex plugins; likely this won't
be used in the near future, but it's readily available if needed

So to confirm you pick one or the other. Can't combine?

File: app/assets/javascripts/admin/addon/components/admin-plugin-config-page.gjs (hunk 1 above)
Comment (martin-brennan) [reply app/assets/javascripts/admin/addon/components/admin-plugin-config-page.gjs:6]: Yeah it kinda irks me each time...but one is an ember template helper (e.g. `{{i18n "blah"}}`) and one is a general `I18n` library that can be used in JS. Will open a dev topic about it, maybe there is something that can be done? Like `import { I18n, i18nHelper } from "discourse-i18n"`...or something

**Comment by martin-brennan** [comment]
@Drenmi correct you only choose one, and I am guessing for the forseeable future only Top will be used. Really depends on how complex a plugin gets.

File: app/assets/javascripts/admin/addon/components/admin-plugin-config-page.gjs (hunk 2 above)
Comment (martin-brennan) [reply app/assets/javascripts/admin/addon/components/admin-plugin-config-page.gjs:11]: Fair enough lol. I'm open to suggestions, but its kinda quite a specific thing. Maybe just `adminPluginNavManager`?

Comment (martin-brennan) [reply app/assets/javascripts/admin/addon/components/admin-plugin-config-page.gjs]: Will go with that

File: app/assets/javascripts/admin/addon/components/admin-plugin-config-page.gjs (hunk 1 above)
Comment (CvX) [reply app/assets/javascripts/admin/addon/components/admin-plugin-config-page.gjs:6]: the helper is just a function that calls `I18n.t()` so you can use just that 🙂

**Pull request merged**